src/new/wallet.js

```diff
@@ -692,153 +869,29 @@ class HathorWallet extends EventEmitter {
+        reject(err);
       });
-    }
-
-    const fullTxData = Object.assign({tokens}, getDefaultData());
-
-    const walletData = wallet.getWalletData();
-    const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
-    const tokensUids = tokens.map((token) => { return {uid: token} });
-    for (const tokenUid in tokensData) {
-      // For each token key in tokensData we prepare the data
-      const partialData = tokensData[tokenUid];
-      let chooseInputs = true;
-      if (partialData.inputs.length > 0) {
-        chooseInputs = false;
-      }
-
-      // Warning: prepareSendTokensData(...) might modify `partialData`. It might add inputs in the inputs array
-      // if chooseInputs = true and also the change output to the outputs array, if needed.
-      // it's not a problem to send the token without the symbol/name. This is used only for error message but
-      // it will increase the complexity of the parameters a lot to add the full token in each output/input.
-      // With the wallet service this won't be needed anymore, so I think it's fine [pedroferreira 04-19-2021]
-      const ret = wallet.prepareSendTokensData(partialData, {uid: tokenUid}, chooseInputs, historyTxs, tokensUids, newOptions);
-
-      if (!ret.success) {
-        ret.debug = {
-          balance: this.getBalance(tokenUid),
-          partialData: partialData, // this might not be the original `partialData`
-          tokenUid,
-          ...ret.debug
-        };
-        return ret;
-      }
-
-      fullTxData.inputs = [...fullTxData.inputs, ...ret.data.inputs];
-      fullTxData.outputs = [...fullTxData.outputs, ...ret.data.outputs];
-    }
-
-    let preparedData = null;
-    try {
-      preparedData = transaction.prepareData(fullTxData, pin);
-      return this.sendPreparedTransaction(preparedData, newOptions);
-    } catch(e) {
-      const message = helpers.handlePrepareDataError(e);
-      return {success: false, message};
-    }
-  }
+    });
 
-  /**
-   * Send a full prepared transaction
-   * Just transform data object to bytes, then hexadecimal and send it to full node.
-   *
-   * @param {Object} data Full transaction data
-   *
-   * @return {Object} Object with {success: true, sendTransaction, promise}, where sendTransaction is a
-   * SendTransaction object that emit events while the tx is being sent and promise resolves when the sending is done
-   **/
-  sendPreparedTransaction(data, options = { startMiningTx: true }) {
-    storage.setStore(this.store);
-    const { startMiningTx } = options;
-    const sendTransaction = new SendTransaction({data});
-    if (startMiningTx) {
-      sendTransaction.start();
-    }
-    const ret = {success: true, promise: sendTransaction.promise, sendTransaction};
-    if (this.debug) {
-      ret.debug = {
-        balanceHTR: this.getBalance(),
-        data: data,
-      };
-    }
-    return ret;
+    sendTransaction.run();
```

Comment:
Should we return a promise in the `run()` method? So we can return it (instead of creating a new one).

Reply:
Done here https://github.com/HathorNetwork/hathor-wallet-lib/pull/221/commits/43cdd56eebf3e443132890aaddd495d24aaf24f5

Reply:
That's great! Thanks!